wordpress-mobile/WordPress-iOS
# Comment Detail: Navigation bar styling

Refs #17087

This PR adds some style ✨ to the navigation bar, according to the designs in Zeplin. I noticed that the look feels similar to Apple's Messages app – where the navigation bar is opaque when positioned on top/scroll edge, but provides visual context for scrolled content behind the navigation bar through a visual blur effect. Here are some previews:

• | Light 🌝 | Dark 🌚 
--|--|--
Top | ![light_top](https://user-images.githubusercontent.com/1299411/135664027-aa0b309d-bf7d-4237-a2eb-ef56f51089c2.png) | ![dark_top](https://user-images.githubusercontent.com/1299411/135664032-671045b2-80f2-4146-8e75-69fff56e2f40.png)
Scrolled | ![light_scrolled](https://user-images.githubusercontent.com/1299411/135664029-59a32648-55ff-40f6-93cb-87f1969d4d43.png) | ![dark_scrolled](https://user-images.githubusercontent.com/1299411/135664014-d6ba4c55-e5e2-4880-aab3-b9811149a827.png)

Additionally, extracted out the assignment of the edit bar button item to its own method, `configureEditButtonItem()` since user permission may change and affect the ability of the user to edit the comment. Moving this to its own method allows it to be called in `refreshData`, so the edit button is hidden/shown correctly based on the newest state.

## To Test

Ensure that the `New Comment Detail` is enabled.

- Go to My Site > Comments.
- Select any comment, preferably one that contains a lot of text (so the view is scrollable). 
- Verify that the navigation bar is opaque when in the top position.
- Scroll the view.
- Verify that the navigation bar shows a visual blur effect.

Ensure that ☝️ behaves as expected in light and dark mode.

## Regression Notes
1. Potential unintended areas of impact
n/a, the feature is still hidden behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a, the feature is still hidden behind a feature flag.

3. What automated tests I added (or what prevented me from doing so)
n/a, the feature is still hidden behind a feature flag.

PR submission checklist:
